jaxb/src/main/java/com/baeldung/jaxb/dateunmarshalling/JaxbDateUnmarshalling.java

```diff
@@ -0,0 +1,52 @@
+package com.baeldung.jaxb.dateunmarshalling;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import java.io.InputStream;
+
+public class JaxbDateUnmarshalling {
+
+    public static final String DEFAULT_DATE_UNMARSHALLING_FILE = "default-date-unmarshalling.xml";
+    public static final String CUSTOM_DATE_UNMARSHALLING_FILE = "custom-date-unmarshalling.xml";
+
+    public static Book unmarshalDates(InputStream inputFile) throws JAXBException {
+        JAXBContext jaxbContext = JAXBContext.newInstance(Book.class);
+        Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
+        Book book = (Book) jaxbUnmarshaller.unmarshal(inputFile);
+        return book;
+    }
+
+    public static Book2 unmarshalDatesUsingCustomXmlAdapter(InputStream inputFile) throws JAXBException {
+        JAXBContext jaxbContext = JAXBContext.newInstance(Book2.class);
+        Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
+        Book2 book = (Book2) jaxbUnmarshaller.unmarshal(inputFile);
+        return book;
+    }
+
+    public static Book3 unmarshalDatesUsingJava8(InputStream inputFile) throws JAXBException {
+        JAXBContext jaxbContext = JAXBContext.newInstance(Book3.class);
+        Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
+        Book3 book = (Book3) jaxbUnmarshaller.unmarshal(inputFile);
+        return book;
+    }
+
+    public static InputStream getInputStream(String file) {
+        ClassLoader classLoader = JaxbDateUnmarshalling.class.getClassLoader();
+        return classLoader.getResourceAsStream(file);
+    }
+
+    public static void main(String[] args) {
+        try {
+            Book book = unmarshalDates(getInputStream(DEFAULT_DATE_UNMARSHALLING_FILE));
+            Book2 book2 = unmarshalDatesUsingCustomXmlAdapter(getInputStream(CUSTOM_DATE_UNMARSHALLING_FILE));
+            Book3 book3 = unmarshalDatesUsingJava8(getInputStream(CUSTOM_DATE_UNMARSHALLING_FILE));
+            System.out.println(book);
+            System.out.println(book2);
+            System.out.println(book3);
+        } catch (JAXBException e) {
+            e.printStackTrace();
```

Comment:
This is IDE generated code... could we avoid the try/catch here and just let `main` throw an Exception?

Or would be better illustrating all this using unit tests, rather than a main method?